### DISCUSSION TITLE
Possible to call Edge Function with pg_net?

### DISCUSSION THREAD

**Discussion by spoken-angus** [supabase/supabase#9955] [Questions] [answered]
Is it possible to call a supabase edge function with pg_net? I'd like to call an edge function after google oauth and was planning to use a database trigger to call it.
I've enabled pg_net in the sql editor
`create schema if not exists net;`
`create extension pg_net with schema net;`

And tried making a trigger to call the edge function:
```
create or replace function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
declare
  payload text;
begin
  payload := '{
      "email": "'||(new.email)||'",
      "id": "'||(new.id)||'"
    }';

  select
    net.http_post(
        url:='https://project-url.functions.supabase.co/get-info',
        body:=payload::jsonb,
        headers:='{"Content-Type": "application/json", "Authorization": "SUPABASE_ANON_KEY"}'::jsonb
    )
  as request_id;

  return new;
end;
$$;
-- trigger the function every time a user is created
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

But when I try logging in with google, I get a database error saving the new user.

**Comment by laktek** [comment]
Can you verify the function works by calling it directly? Also, do you see any errors on the function's logs?

**Reply by spoken-angus** [reply]
The function does work by calling it directly. Additionally, there are no errors in the edge function logs since it never even gets invoked

**Comment by spoken-angus** [comment, marked as answer]
Could it be an issue with the way I've set up the pg_net extension?

When I try the example in the API reference I get an error: `An error has occured: Failed to fetch`

```
select
    net.http_post(
        url:='https://httpbin.org/post',
        body:='{"hello": "world"}'::jsonb
    ) as request_id;

select * from net.http_collect_response(3, async:=false);
```

**Reply by steve-chavez** [reply]
BLUF: don't use the `net.http_collect_response` function, if you want a sync response you should use [pgsql-http](https://github.com/pramsey/pgsql-http/)

The `net.http_collect_response`([ref](https://github.com/supabase/pg_net/blob/master/sql/pg_net.sql#L236-L257)) calls up the [net._await_response](https://github.com/supabase/pg_net/blob/master/sql/pg_net.sql#L48-L74) which loops over the `net._http_response` table and locks it, preventing the pg_net worker from progressing.

`http_collect_response` is a private function only used for testing, it will be marked as so and fixed on `pg_net`.

**Reply by steve-chavez** [reply]
`pg_net` is working again on your instance after restarting it, you can try other requests now.

**Comment by steve-chavez** [comment]
Can you check the `net._http_response` table for more details on the errors? Do

```sql
select * from net._http_response;
```

Also check you're on the latest version of pg_net(0.6) with

```sql
select * from pg_extension where extname = 'pg_net';
```

**Reply by spoken-angus** [reply]
It looks like I'm on the latest version of pg_net, but `select * from net._http_response;` doesn't return any rows.

**Reply by steve-chavez** [reply]
> It looks like I'm on the latest version of pg_net

You can request an upgrade by emailing to support@supabase.com.

**Reply by spoken-angus** [reply]
I'm on version 0.6, do I still need an upgrade?

**Reply by steve-chavez** [reply]
Sorry, I misread your previous comment. No upgrade would be needed then.

> but select * from net._http_response; doesn't return any rows.

There's a TTL for the `_http_response` table. You'd need to retry the requests to see the errors.

**Reply by spoken-angus** [reply]
Yeah I tried rerunning some requests, and it still doesn't return any rows.

**Reply by steve-chavez** [reply]
Can you send your project ref to support@supabase.com?

**Reply by spoken-angus** [reply]
sent!
